boot_shutdown_manager/src/boot_shutdown_manager_core.cpp

```diff
@@ -274,25 +310,38 @@ bool BootShutdownManager::isReady() const
 void BootShutdownManager::executeShutdown()
 {
   RCLCPP_INFO(get_logger(), "ExecuteShutdown start");
-  rclcpp::Time latest_power_off_time = now();
-  auto req = std::make_shared<ExecuteShutdown::Request>();
+  rclcpp::Clock clock(RCL_SYSTEM_TIME);
+  rclcpp::Time latest_power_off_time = clock.now();
+
   for (auto & [ecu_name, client] : ecu_client_queue_) {
     if (client->skip_shutdown) {
       continue;
     }
-    auto resp = call<ExecuteShutdown>(client->cli_execute, req);
-    if (!resp) {
-      RCLCPP_WARN(get_logger(), "[%s] ExecuteShutdown service call faild.", ecu_name.c_str());
-      // Prevent process ressart due to null access when ECU is not running
-      continue;
-    }
-    rclcpp::Time power_off_time = resp->power_off_time;
-    if (latest_power_off_time < power_off_time) {
-      latest_power_off_time = power_off_time;
+
+    try {
+      ExecuteShutdownService req;
+      auto resp = client->cli_execute->call(req);
```

Comment:
"There's a possibility that resq is [undefined](https://github.com/tier4/boot_shutdown_tools/blob/fa4df749f9b45e5cd2759596d6ac52577e748fcd/boot_shutdown_communication/include/boot_shutdown_communication/service_client.hpp#L116). It is not checked, but is it okay?"

Reply:
It' not ok, Thanks!

Reply:
@TetsuKawa 
The function does not return undefined values since it uses value passing, not pointer passing.
but I changed it to return a pointer for easier error handling. Please check these fixes.
- The branch feat/non-ros-communication-interface
  https://github.com/tier4/boot_shutdown_tools/pull/27/commits/22a12120e77b1f0c3c6fe9e80d0851743b34e405
- This branch
  https://github.com/tier4/boot_shutdown_tools/pull/30/commits/2987218fa3433c6ee22b353eda670a3c0ebafedf